internal/pkg/agent/install/uninstall.go

```diff
@@ -181,11 +189,30 @@ func containsString(str string, a []string, caseSensitive bool) bool {
 }
 
 func uninstallComponents(ctx context.Context, cfgFile string, uninstallToken string, pt *progressbar.ProgressBar) error {
-	log, err := logger.NewWithLogpLevel("", logp.ErrorLevel, false)
+	var err error
+	logCfg := logp.DefaultConfig(logp.DefaultEnvironment)
+	// Using in memory logger, so we don't write logs to the
+	// directory we are trying to delete
+	logp.ToObserverOutput()(&logCfg)
+
+	err = logp.Configure(logCfg)
```

Comment:
I believe this is changing a global. I think it would be better to set this in the root of the uninstall command and pass the logger into this function. That would ensure that if the logger is used anywhere else in the future of the uninstall command that it would always use this logger.

Reply:
made the change, take a look and see if that is what you were expecting.